contracts/contracts/token/OUSD.sol

```diff
@@ -445,50 +434,61 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
         view
         returns (uint256)
     {
-        if (nonRebasingCreditsPerToken[_account] != 0) {
-            return nonRebasingCreditsPerToken[_account];
+        if (alternativeCreditsPerToken[_account] != 0) {
+            return alternativeCreditsPerToken[_account];
         } else {
             return _rebasingCreditsPerToken;
         }
     }
 
     /**
-     * @dev Is an account using rebasing accounting or non-rebasing accounting?
-     *      Also, ensure contracts are non-rebasing if they have not opted in.
+     * @dev Before a `rebaseOptIn` or non yield delegating token `transfer` can be executed contract
+     *      accounts need to have a more explicitly defined rebasing state set.
+     *
+     *      Contract account can be in the following states before `autoMigrate` is called:
+     *      1. Under any token contract codebase they haven't been part of any token transfers yet
+     *         having rebaseState `NotSet` and `alternativeCreditsPerToken == 0`
+     *      2. Under older token contract codebase they have the default rebaseState set to `NotSet` and
+     *         the codebase has "auto-migrated" them by setting the `alternativeCreditsPerToken` to some
+     *         value greater than 0.
+     *      3. Contract has under any token contract codebase explicitly requested to be opted out of rebasing
+     *
+     *     Case 1. Needs to be migrated using autoMigrate to a nonRebasing account.
+     *
+     *     Note: Even with this _autoMigrate function in place there will still be Case 2 accounts existing that
+     *           will behave exactly like RebaseState StdNonRebasing account, and still having their rebase state
+     *           set to `NotSet`
+     *
      * @param _account Address of the account.
      */
-    function _isNonRebasingAccount(address _account) internal returns (bool) {
-        bool isContract = Address.isContract(_account);
-        if (isContract && rebaseState[_account] == RebaseOptions.NotSet) {
-            _ensureRebasingMigration(_account);
+    function _autoMigrate(address _account) internal returns (bool) {
```

Comment:
I don't know if you have easy gas measuring setup, but I wonder if this would be gas cheaper if it took the rebaseState as a parameter. We know that rebaseState will have been loaded from storage by the parent function already. This would mean that this function does not do a separate load.

Reply:
I don't have a way to single out one call to measure the gas usage of storage slots. So I am running a single test that because of the fixtures makes some extra transfer calls. The difference passing in the rebase state is minimal when it comes to gas consumption. Non optimised: 
![Screenshot 2024-11-18 at 23 34 50](https://github.com/user-attachments/assets/795d218d-f487-4b00-82d3-fb9a01763355)

Gas optimised: 
![Screenshot 2024-11-18 at 23 33 30](https://github.com/user-attachments/assets/bc13899f-b314-4ef8-8589-90e181b0e681)

It seems to conforms to [EIP 2929](https://eips.ethereum.org/EIPS/eip-2929#parameters) where warm reads of the storage slot net an extra 100 gas cost per load as opposed to 2100 gas cost of accessing a cold storage slot. 

IMO that doesn't justify the code changes.

Reply:
Good keeping as is then.